content/inferno/time-series/time-series.md

```diff
@@ -0,0 +1,309 @@
+:::
+
+As usual, Virgilio has collected for you the best free resources available, let's see!
+
+The first and most important resource in time series forecasting is the following textbook:
+
+- [**Forecasting: Principles And Practice**](https://otexts.com/fpp2/)
+
+The textbook uses R examples throughout the lessons, but it's not mandatory to learn R to use it! 
+
+The advice Virgilio gives you is to try to follow the lessons **translating** the R code to Python code, obviously helping you with Google and all the resources at your disposal!
+
+In this way, you'll learn a lot about how to use Python for time series forecasting, and you'll have a lot of **reusable code** for your future projects!
+
+Some other useful resources are:
+
+- [Applying Statistical Modeling & Machine Learning to Perform Time-Series Forecasting](https://www.youtube.com/watch?v=JntA9XaTebs&t=6s&ab_channel=PyData)
+- [A Worked Example of Using Neural Networks for Time Series Prediction](https://www.youtube.com/watch?v=hAlGqT3Xpus&ab_channel=PyData)
+- [Reliably forecasting time-series in real-time ](https://www.youtube.com/watch?v=0zpg9ODE6Ww&ab_channel=PyData)
+- [Time Series Forecasting using Statistical and Machine Learning Models](https://www.youtube.com/watch?v=_vQ0W_qXMxk&ab_channel=PyData)
+
+::: tip
+It's important to understand that no-one-method will always outperform others, and that time series forecasting is a hot research topic, and you should always try to be aware of new techniques or approaches! 
+
+Be sure to read some survey papers, once a while, like [**this one**](https://arxiv.org/abs/2004.13408)!
+:::
+
+Check also the [**Prophet**](https://facebook.github.io/prophet/) project, from Facebook AI Labs!
+
+Prophet is a procedure for forecasting time series data based on an additive model where non-linear trends are fit with yearly, weekly, and daily seasonality, plus holiday effects. 
+
+It works best with time series that have strong seasonal effects and several seasons of historical data. 
+
+Prophet is robust to missing data and shifts in the trend, and typically handles outliers well.
+
+
+## Time Series Forecasting as a Classification Problem
+
+A very useful approach to keep in mind when working with time series forecasting is to _treat the problem of forecasting as a classification problem._
+
+If treated in this way, the problem can be "simplified" and so be faced with a simpler problem to solve.
+
+For example, **preventing the exact price** of Apple shares tomorrow could be a very difficult challenge, but fortunately, it is a problem that can be _"simplified"_!
+
+An effective way to simplify this problem is to divide the space of tomorrow's possible prices (in terms of % variation) into bins! 
+
+For example with 20 bins, the classes would be: [-100%, -90%], [-90%,-80%]....the other bins...[+80%, +90%], [+90%,+100%].
```

Comment:
Also called Histogram binning (you can use the `qcut` and `cut` features of the `Pandas` `DataFrame` to achieve this)